File: io-hotmoka-network/src/main/java/io/hotmoka/network/models/signatures/FieldSignatureModel.java
package io.hotmoka.network.models.signatures;

import io.hotmoka.beans.annotations.Immutable;
import io.hotmoka.beans.signatures.FieldSignature;

/**
 * The model of the signature of a field of a class.
 */
@Immutable
public final class FieldSignatureModel extends SignatureModel {

	/**
	 * The name of the field.
	 */
	public final String name;

	/**
	 * The type of the field.
	 */
	public final String type;

	/**
	 * Builds the model of the signature of a field.
	 * 
	 * @param field the signature of the field
	 */
	public FieldSignatureModel(FieldSignature field) {
		super(field.definingClass.name);

		this.name = field.name;
		this.type = nameOf(field.type);
	}

	/**
	 * Yields the signature having this model.
	 * 
	 * @return the signature
	 */
	public FieldSignature toBean() {
		return new FieldSignature(definingClass, name, typeWithName(type));
	}
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/signatures/FieldSignatureModel.kt
package io.hotmoka.network.thin.client.models.signatures

/**
 * The model of the signature of a field of a class.
 */
class FieldSignatureModel(
        /**
         * The name of the field.
         */
        val name: String,
        /**
         * The type of the field.
         */
        val type: String,
        definingClass: String
): SignatureModel(definingClass)